### PR TITLE
"Fixes" the rest of the floating promises.

### PR DESCRIPTION
**Pull request by csilvestrini**
Also enables the no-floating-promises lint check.

Some of the floating promises I was able to fix properly, but a few had
to be whitelisted via a new ignorePromiseHack() function. Someone who
knows what the code is doing needs to debug these...

Reduces number of floating promises from 14 -> 0 (worth 1.4 points) #paydown

### CODE REVIEW

File: src/runtime/util.ts [hunk 1]
```diff
@@ -66,3 +66,11 @@ export function setDiffCustom<T, U>(from: T[], to: T[], keyFn: (T) => U): {add:
   }
   return result;
 }
+
+/**
+ * A hack to ignore a floating promise and bypass the linter. Promises should *not* be ignored. They should always be awaited or returned. See
```
Comment (piotrswigon) [inline src/runtime/util.ts]: I would rather say promises should rarely be left floating and when such behaviour is intended, they should be clearly marked as such.

File: src/runtime/util.ts [hunk 2]
```diff
@@ -66,3 +66,11 @@ export function setDiffCustom<T, U>(from: T[], to: T[], keyFn: (T) => U): {add:
...
+ * https://tsetse.info/must-use-promises.html for details.
+ *
+ * TODO: Remove all usages of this function and then delete it.
+ */
+export function ignorePromiseHack<T>(promise: Promise<T>) {}
```
Comment (piotrswigon) [inline src/runtime/util.ts]: I would go for something along the lines of `floatingPromiseToAudit()`, as I don't find `ignorePromiseHack` as descriptive, but feel free to ignore me.

File: config/tslint.base.json [hunk 3]
```diff
@@ -31,6 +31,7 @@
     "no-debugger": true,
     "no-default-export": true,
     "no-duplicate-variable": true,
+    "no-floating-promises": true,
```
Comment (piotrswigon) [inline config/tslint.base.json]: Woohoo!

File: src/runtime/util.ts (hunk 1 above)
Comment (csilvestrini) [reply src/runtime/util.ts]: done

File: src/runtime/util.ts (hunk 2 above)
Comment (csilvestrini) [reply src/runtime/util.ts]: I like it too. done

**Pull request merged**